aspnet/dnx/commands.rst

```diff
@@ -1,8 +1,232 @@
-.. include:: /../common/stub-topic.txt
+Using Commands
+==============
 
-|stub-icon| Using Commands
-==========================
+DNX projects are used to build and run .NET applications for Windows, Mac and Linux. DNX provides a host process, CLR hosting logic and managed entry point discovery. You can use DNX to execute commands from a command prompt. 
 
-.. include:: /../common/stub-notice.txt
-    
-.. _issue: https://github.com/aspnet/Docs/issues/55
+What is a command?
+------------------
+
+A command is a named execution of a .NET entry point with specific arguments. Commands can be defined locally in your project or installed globally on your machine. The *project.json* file in your project allows you to define commands for your project. Commands that you define for your projects are understood by Visual Studio Code (VS Code) as well as Visual Studio. Global commandsw, which are not project specific, can be installed on a machine and run from a command prompt.
+
+Using commands in your project
+------------------------------
+
+In the ``commands`` section of the below *project.json* example, four commands are listed:
+
+.. code-block:: json
+
+    {
+      "version": "1.0.0",
+      "webroot": "wwwroot",
+      "exclude": [
+        "wwwroot"
+      ],
+      "dependencies": {
+        "Kestrel": "1.0.0-beta6",
+        "Microsoft.AspNet.Diagnostics": "1.0.0-beta6",
+        "Microsoft.AspNet.Hosting": "1.0.0-beta6",
+        "Microsoft.AspNet.Server.IIS": "1.0.0-beta6",
+        "Microsoft.AspNet.Server.WebListener": "1.0.0-beta6",
+        "Microsoft.AspNet.StaticFiles": "1.0.0-beta6"
+      },
+      "commands": {
+		"web": "Microsoft.AspNet.Hosting --config hosting.ini",
+		"kestrel": "Microsoft.AspNet.Hosting --config hosting.ini",
+		"gen": "Microsoft.Framework.CodeGeneration",
+		"ef": "EntityFramework.Commands"
+      },
+      "frameworks": {
+        "dnx451": { },
+        "dnxcore50": { }
+      }
+    }
+
+A command can include a set of arguments that will be passed to the DNX. In the above example, the first part of a command statement is an assembly with an entry point that the DNX will try to execute. Notice that in the ``commands`` section shown above, the ``ef`` command is implemented by the ``EntityFramework.Commands`` assembly. This command doesn't require any extra argument, all that is needed to define the command is the name of the assembly. For the ``web`` command and the ``kestrel`` command, the arguments are contained in the referenced *hosting.ini* file. In the *hosting.ini* file you will see the ``server`` agrument and the the ``server.urls`` argument. The ``kestrel`` command, as well as the ``web`` command, will check the ``Microsoft.AspNet.Hosting`` assembly for an entry point, then it will pass the ``server`` and ``serveral.urls`` agruments to the entry point. 
```

Comment:
I would mention here that the extra arguments to each command are passed to the entry point through the `args` argument of the `main` method